modeshape-jcr/src/main/java/org/modeshape/jcr/index/local/ManagedLocalIndex.java

```diff
@@ -85,4 +93,17 @@ public void removeAll() {
         index.removeAll();
     }
 
+    @Override
+    public IndexManager.IndexStatus getStatus() {
+        return status.get();
+    }
+
+    @Override
+    public void updateStatus(IndexManager.IndexStatus currentStatus, IndexManager.IndexStatus newStatus) {
+        this.status.compareAndSet(currentStatus, newStatus);
```

Comment:
Well, my concern is that the state gets left in some unusable state (e.g., reindexing) when it is no longer at that state, simply because the compare part of the `compareAndSet` failed. 

I'm not overly concerned, and I'm fine if you want to leave it as is.